Manuscript

wincowgerDEV opened the pull request:
Hoping all coauthors can review the manuscript by September 19th. Let me know if you need more time.

wincowgerDEV committed:
my workflow

wincowgerDEV committed:
example templates for manuscript.

wincowgerDEV committed:
manuscript info

wincowgerDEV committed:
update citations

wincowgerDEV committed:
cleanup text

wincowgerDEV commented:
I would like to add two other coauthors to the manuscript from Let's Do It World (@KristiinaKerge and @krishaamer). They were a huge part of the initial push for [WADE AI](https://github.com/letsdoitworld/wade-ai) which inspired this project and they are interested in helping us review the manuscript and help us make the tool have a global presence.

wincowgerDEV committed:
cite

wincowgerDEV committed:
update readme

wincowgerDEV committed:
contribute

wincowgerDEV committed:
add website badge

wincowgerDEV committed:
update web link

wincowgerDEV committed:
Update README.md

wincowgerDEV committed:
Update README.md

wincowgerDEV committed:
add video link

wincowgerDEV committed:
auto update

wincowgerDEV committed:
Merge branch 'manuscript' of https://github.com/code4sac/trash-ai into manuscript

wincowgerDEV committed:
utility files device specific

wincowgerDEV committed:
move the pdf generator to workflows

wincowgerDEV committed:
test error in paper.bib

wincowgerDEV committed:
test citation style

wincowgerDEV committed:
add citations

wincowgerDEV committed:
add a few more citations

wincowgerDEV committed:
try without space

wincowgerDEV committed:
add all

wincowgerDEV committed:
no spaces allowed in citation names

wincowgerDEV committed:
add demo images

wincowgerDEV committed:
Update paper.md

wincowgerDEV committed:
give images some space

wincowgerDEV committed:
image captions

wincowgerDEV committed:
add mention of wade

wincowgerDEV committed:
update authors and acknowledgements

feydan commented on paper.md:
```diff
@@ -0,0 +1,111 @@
+
+# Statement of need
+
+The trash in the environment is a widespread problem that is difficult to measure. Classical measurement techniques require surveyors with pen and paper to manually quantify every piece of trash at a site. This method is time-consuming. Scientists are actively trying to address this issue by using imaging to better understand the prevalence and distribution of trash in an `efficient yet effective ...
```
> average data scientist

should this be `littler researcher`?  I feel like data scientists would be the ones who know how to run models.

> and serverless architecture (e.g. AWS Lambda)

Currently lambda isn't really doing anything special or enabling any workflows that wouldn't be possible with other app hosting methods.  We are primarily using it right now because it is currently the cheapest way to host our simple back-end application (most of the logic is on the front-end).

wincowgerDEV replied:
Updated to litter researcher, good idea. 

Added the word affordable for the second part, so that it is clear that is why we are going serverless.

wincowgerDEV replied:
https://github.com/code4sac/trash-ai/commit/4ff318f9c429d80adf701fee7131f2329e93818d

feydan commented on paper.md:
```diff
@@ -0,0 +1,111 @@
+# Method
+
+## Workflow Overview
+Trash AI is trained on the [TACO dataset](http://tacodataset.org/) using [YOLO 5](pytorch.org). Trash AI stores images in [IndexDB](https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API) to keep the data primarily browser side and uses [tensorflow.js](https://www.tensorflow.org/js) to keep analysis browser side too. When images are uploaded to the browser, Trash AI provides the predictio ...
```
> Any data uploaded to the platform is automatically saved to an

may be automatically saved   -- I think this is currently disabled, but we may re-enable it in the future.

wincowgerDEV replied:
updated with your recs https://github.com/code4sac/trash-ai/commit/4ff318f9c429d80adf701fee7131f2329e93818d

feydan commented on paper.md:
```diff
@@ -0,0 +1,111 @@
+From our experience, the accuracy of the model varies depending on the quality of the images and their context/background. Trash is a nuanced classification because the same object in different settings will not be considered trash (e.g. a drink bottle on someone's desk vs in the forest laying on the ground). This and other complexities to trash classification make a general trash AI a challengin ...
+
+# Availability
+Trash AI is hosted on the web at www.trashai.org. The source code is [available on github](https://github.com/code4sac/trash-ai) with an [MIT license](https://mit-license.org/). The source code can be run offline on any machine that can install [Docker and Docker-compose](www.docker.com). [Documentation](https://github.com/code4sac/trash-ai#ai-for-litter-detection-web-application) is maintained b ...
```
Code for Sacramento and Open Fresno

wincowgerDEV replied:
added https://github.com/code4sac/trash-ai/commit/4ff318f9c429d80adf701fee7131f2329e93818d

feydan commented on paper.md:
```diff
@@ -0,0 +1,111 @@
+From our experience, the accuracy of the model varies depending on the quality of the images and their context/background. Trash is a nuanced classification because the same object in different settings will not be considered trash (e.g. a drink bottle on someone's desk vs in the forest laying on the ground). This and other complexities to trash classification make a general trash AI a challengin ...
+
+# Availability
+Trash AI is hosted on the web at www.trashai.org. The source code is [available on github](https://github.com/code4sac/trash-ai) with an [MIT license](https://mit-license.org/). The source code can be run offline on any machine that can install [Docker and Docker-compose](www.docker.com). [Documentation](https://github.com/code4sac/trash-ai#ai-for-litter-detection-web-application) is maintained b ...
```
Might also be worth mentioning that there are some instructions (though not exhaustive) for [setting it up and hosting it via your own AWS account](https://github.com/code4sac/trash-ai/blob/manuscript/docs/git-aws-account-setup.md) if anyone was interested in forking, modifying, and/or white labeling the application.  Our deployment model also [uses github actions](https://github.com/code4sac/trash-ai/blob/manuscript/docs/github-actions.md), so the CI/CD setup is built into the code.  These concerns are non-trivial to solve for any team that is looking maintain a project like this with multiple contributors.

wincowgerDEV replied:
added https://github.com/code4sac/trash-ai/commit/4ff318f9c429d80adf701fee7131f2329e93818d

feydan commented on paper.md:
```diff
@@ -0,0 +1,111 @@
+This workflow is likely to be highly useful for a wide variety of computer vision applications and we hope that people reuse the code for applications beyond trash detection. We aim to increase the labeling of images by creating a user interface that allows users to improve the annotations that the model is currently predicting by manually restructuring the bounding boxes and relabeling the class ...
+
+# Acknowledgements
+Code for Sacramento led the development of the software tool. The Moore Institute advised on priorities and led the drafting of this manuscript. Let's Do It Foundation assisted with original products leading up to trash AI in the development of WADE AI. We acknowledge the work of the Code for Sacramento and Fresno team, part of code for America, without whom this project would not have been possi ...
```
Code for Sacramento and Open Fresno

wincowgerDEV replied:
added https://github.com/code4sac/trash-ai/commit/4ff318f9c429d80adf701fee7131f2329e93818d

feydan commented on paper.md:
```diff
@@ -0,0 +1,111 @@
+This workflow is likely to be highly useful for a wide variety of computer vision applications and we hope that people reuse the code for applications beyond trash detection. We aim to increase the labeling of images by creating a user interface that allows users to improve the annotations that the model is currently predicting by manually restructuring the bounding boxes and relabeling the class ...
+
+# Acknowledgements
+Code for Sacramento led the development of the software tool. The Moore Institute advised on priorities and led the drafting of this manuscript. Let's Do It Foundation assisted with original products leading up to trash AI in the development of WADE AI. We acknowledge the work of the Code for Sacramento and Fresno team, part of code for America, without whom this project would not have been possi ...
```
We acknowledge the work of the Code for Sacramento and Open Fresno teams, brigades of Code for America

^ Code for Sacramento and Open Fresno are not actually legally a 'part' of Code for America.  We are separate legal entities that have a partnership with Code for America via a Memorandum of Understanding.

wincowgerDEV replied:
added https://github.com/code4sac/trash-ai/commit/4ff318f9c429d80adf701fee7131f2329e93818d

feydan commented on paper.md:
```diff
@@ -0,0 +1,111 @@
+affiliations:
+ - name: Moore Institute for Plastic Pollution Research, USA
+   index: 1
+ - name: Code for Sacramento and Fresno, USA
```
Code for Sacramento and Open Fresno

wincowgerDEV replied:
added https://github.com/code4sac/trash-ai/commit/4ff318f9c429d80adf701fee7131f2329e93818d

feydan commented on paper.md:
```diff
@@ -0,0 +1,111 @@
+    affiliation: 4
+  - name: Kris Haamer
+    affiliation: 4
+  - name: Gina 
```
Gina Durante

wincowgerDEV replied:
added https://github.com/code4sac/trash-ai/commit/4ff318f9c429d80adf701fee7131f2329e93818d

feydan commented on paper.md:
```diff
@@ -0,0 +1,111 @@
+    affiliation: 4
+  - name: Gina 
+    affiliation: 2
+  - name: Brianda
```
Brianda Hernandez

wincowgerDEV replied:
added https://github.com/code4sac/trash-ai/commit/4ff318f9c429d80adf701fee7131f2329e93818d

wincowgerDEV committed:
Updated with Dan's Recs

shollingsworth commented on paper.md:
```diff
@@ -0,0 +1,111 @@
+  - Litter
+  - AI
+  - Image Classification
+  - Serverless
```
Since we're discussing specific tech I'd suggest adding
* vue
* vuetify
* vite
* pinia

wincowgerDEV replied:
Thanks @shollingsworth got these incorporated. https://github.com/code4sac/trash-ai/commit/3652a152ea3d3afb7e6df823f0791b1bb0e081a4

shollingsworth reviewed (approved):
looks good 👍🏻

walteryu reviewed (approved):
Manuscript looks great - my only comment is for the limitations section: One limitation based on user feedback is the distance of trash from the camera. Since the AI model was trained using the TACO dataset which consists of images of trash within a reasonable distance, it performs less accurately on images with trash which is farther away such as when taken from a vehicle.

maryfnorris commented:
I did notice that Democracy Lab is listed twice. Otherwise,  this all looks good. Thank you for acknowledging Open Fresno.

hernandezforges reviewed (commented):
No comments.

wincowgerDEV committed:
Mary Comments

https://github.com/code4sac/trash-ai/pull/69#issuecomment-1257554556

wincowgerDEV commented:
> I did notice that Democracy Lab is listed twice. Otherwise, this all looks good. Thank you for acknowledging Open Fresno.

Thanks @maryfnorris for the review and for finding that. Correction made.

wincowgerDEV commented:
> No comments.

Thanks @createHernandez for the review and for the positive response.

wincowgerDEV committed:
SteveO comments

https://github.com/code4sac/trash-ai/pull/69#pullrequestreview-1119350215

wincowgerDEV committed:
Walter's comments

https://github.com/code4sac/trash-ai/pull/69#pullrequestreview-1119351559

wincowgerDEV commented:
> Manuscript looks great - my only comment is for the limitations section: One limitation based on user feedback is the distance of trash from the camera. Since the AI model was trained using the TACO dataset which consists of images of trash within a reasonable distance, it performs less accurately on images with trash which is farther away such as when taken from a vehicle.

Thanks Walter, I have added this context into https://github.com/code4sac/trash-ai/commit/39726ed2705bfc3580c8e8dc7289484a85c21e59

wincowgerDEV committed:
Create config.yml

wincowgerDEV committed:
Create bug.yml

wincowgerDEV committed:
Create feature.yml

wincowgerDEV committed:
hyperlink

wincowgerDEV committed:
Update bug.yml

wincowgerDEV committed:
Update feature.yml

maryfnorris commented:
We have the green light to add Brad Andersen as a contributor.

wincowgerDEV commented:
> We have the green light to add Brad Andersen as a contributor.

Great to have Brad on board, can we add him as a contributor so that he can do his review of the manuscript?

wincowgerDEV committed:
Added in Kristiina and Kris's comments

wincowgerDEV committed:
added kristiina and kris comments

Merge branch 'manuscript' of https://github.com/code4sac/trash-ai into manuscript

# Conflicts:
#	paper.md

gdurante2019 commented on README.md:
```diff
@@ -1,27 +1,38 @@
 
--   Description: Develop web application where users can upload photos of street litter which will be labeled using computer vision to detect and categorize litter type.
--   Benefit: Enhance abilities of researchers to quickly label photos and categorize types.
+Trash AI is a web application where users can upload photos of litter which will be labeled using computer vision to detect and categorize litter in the image by type. Early inspiration from [WADE AI](https://github.com/letsdoitworld/wade-ai) streamlined this development. Trash AI will enhance the abilities of researchers to quickly label trash in photos.
```
First sentence--suggest placing comma after "litter" and before "which".

wincowgerDEV replied:
updated! Thanks for the recommendation.

wincowgerDEV committed:
Update README.md

maryfnorris commented:
Absolutely!

Mary C Norris

On Sun, Oct 16, 2022, 3:43 AM Win Cowger, PhD ***@***.***>
wrote:

> We have the green light to add Brad Andersen as a contributor.
>
> Great to have Brad on board, can we add him as a contributor so that he
> can do his review of the manuscript?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/code4sac/trash-ai/pull/69#issuecomment-1279943117>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANN6M6WCEWWE53MQR6BDYIDWDPL6TANCNFSM6AAAAAAQFJB5YA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

gdurante2019 commented on paper.md:
```diff
@@ -0,0 +1,113 @@
+---
+
+# Summary
+Although computer vision classification routines have been created for trash, they have not been accessible to most researchers due to the challenges in deploying the models. Trash AI is a web GUI (Graphical User Interface) for serverless computer vision classification of batch images with trash in them hosted at www.trashai.org. With a single batch upload and download, a user can automatically d ...
```
Second sentence--suggest changing to read as follows:  "...classification of **individual items of trash within images**, hosted at www.trashai.org." (Bolded for emphasis.)  Since the next sentence begins by stating that images can be uploaded as a batch, it's not necessary to state that in the previous sentence, and the original wording was a bit awkward.

wincowgerDEV replied:
https://github.com/code4sac/trash-ai/commit/3a33857c8ff03c2b308c58b3b5a7bda3a3a79ba9

gdurante2019 commented on paper.md:
```diff
@@ -0,0 +1,113 @@
+Although computer vision classification routines have been created for trash, they have not been accessible to most researchers due to the challenges in deploying the models. Trash AI is a web GUI (Graphical User Interface) for serverless computer vision classification of batch images with trash in them hosted at www.trashai.org. With a single batch upload and download, a user can automatically d ...
+
+# Statement of need
+The trash in the environment is a widespread problem that is difficult to measure. Policy makers require high quality data on trash to create effective policies. Classical measurement techniques require surveyors with pen and paper to manually quantify every piece of trash at a site. This method is time-consuming. Scientists are actively trying to address this issue by using imaging to better und ...
```
First sentence:  Delete "The" and begin the sentence with "Trash is..."

Sentence 10--put a comma after each 'e.g.', as follows:  "New developments in browser-side AI (e.g., tensorflow.js) and serverless architecture (e.g., AWS Lambda)"

Sentence 10--Put a comma after 'GUI', as follows:  "...affordable browser-side artificial intelligence in a web GUI, alleviating both obstacles."

Section is excellent otherwise--very clear!

wincowgerDEV replied:
I want to leave in the "In the environment part because there needs to be a distinction between what we are doing and waste management AI which is a whole different problem that I am not sure this addresses.

wincowgerDEV replied:
Made other recommended changes in https://github.com/code4sac/trash-ai/commit/01e6bad6f97ffc431a635a7883476e12524ac3f0

gdurante2019 commented on paper.md:
```diff
@@ -0,0 +1,113 @@
+# Method
+
+## Workflow Overview
+Trash AI is trained on the [TACO dataset](http://tacodataset.org/) using [YOLO 5](pytorch.org). Trash AI stores images in [IndexDB](https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API) to keep the data primarily browser side and uses [tensorflow.js](https://www.tensorflow.org/js) to keep analysis browser side too. When images are uploaded to the browser, Trash AI provides the predictio ...
```
Last sentence--suggest adding comma after link to S3 and before the word 'which', as follows:  "...saved to an [S3 bucket](https://aws.amazon.com/s3/), which we can use to improve the model over time."

wincowgerDEV replied:
Made change https://github.com/code4sac/trash-ai/commit/14eb61cb7ef35ff83a27bfc045718aef1ecbb60e

gdurante2019 commented on paper.md:
```diff
@@ -0,0 +1,113 @@
+Trash AI is trained on the [TACO dataset](http://tacodataset.org/) using [YOLO 5](pytorch.org). Trash AI stores images in [IndexDB](https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API) to keep the data primarily browser side and uses [tensorflow.js](https://www.tensorflow.org/js) to keep analysis browser side too. When images are uploaded to the browser, Trash AI provides the predictio ...
+
+## AI Training
+The AI model was developed starting with the TACO dataset which was available with a complimentary Jupyter Notebook on [Kaggle](https://www.kaggle.com/datasets/kneroma/tacotrashdataset). An example notebook was referenced which used the default `YOLO v5 model` [@Jocher:2020] as the basic model to begin transfer learning. Next, transfer learning was completed using the entire TACO dataset to impor ...
```
Add a comma before the word 'which' in the first and second sentences, as follows:  

First sentence:  "...starting with the TACO dataset, which was available with a complimentary Jupyter Notebook..."

Second sentence:  "...An example notebook was referenced, which used the default `YOLO v5 model`..."

wincowgerDEV replied:
Change made. https://github.com/code4sac/trash-ai/commit/1654df8a4e2224fa744d99d8fcafd30b2587e773

gdurante2019 commented on paper.md:
```diff
@@ -0,0 +1,113 @@
+The AI model was developed starting with the TACO dataset which was available with a complimentary Jupyter Notebook on [Kaggle](https://www.kaggle.com/datasets/kneroma/tacotrashdataset). An example notebook was referenced which used the default `YOLO v5 model` [@Jocher:2020] as the basic model to begin transfer learning. Next, transfer learning was completed using the entire TACO dataset to impor ...
+
+## Limitations
+From our experience, the accuracy of the model varies depending on the quality of the images and their context/background. Trash is a word people use for an object that lacks purpose, and the purpose of an object is often not obvious in an image. Trash is a nuanced classification because the same object in different settings will not be considered trash (e.g. a drink bottle on someone's desk vs i ...
```
Sentence 3--suggest comma after 'e.g.':  "...not be considered trash (e.g., a drink bottle on someone's desk vs in the forest..."

Sentence 6--suggest hyphen for 'long term':  "...make a general trash AI a challenging (yet worthwhile) long-term endeavor."

Sentence 7--suggest rewriting as follows:  "The algorithm is primarily trained on single pieces of trash in the image, with the trash laying on the ground; thus, model class prediction of trash in these kinds of images will generally be better than, say, trash appearing in aerial images."  (Note--I wouldn't say that the current model 'excels' at correctly classifying trash in this or any other use case, but rather, the accuracy of its classifications will be better for these kinds of images than other types of images containing trash.)

wincowgerDEV replied:
Edits made in https://github.com/code4sac/trash-ai/commit/0b4743ed252ad7f7351950aecef4c1c77e8e7aa0. Split up your recommendation for sentence 7 a little so that it wouldn't be as long.

gdurante2019 commented on paper.md:
```diff
@@ -0,0 +1,113 @@
+This workflow is likely to be highly useful for a wide variety of computer vision applications and we hope that people reuse the code for applications beyond trash detection. We aim to increase the labeling of images by creating a user interface that allows users to improve the annotations that the model is currently predicting by manually restructuring the bounding boxes and relabeling the class ...
+
+# Acknowledgements
+Code for Sacramento and Open Fresno led the development of the software tool. The Moore Institute advised on priorities and led the drafting of this manuscript. Let's Do It Foundation assisted with original products leading up to trash AI in the development of WADE AI. We acknowledge the work of the Code for Sacramento and Open Fresno team, brigades of code for America, without whom this project  ...
```
Sentence 4--capitalize 'code' in 'brigades of code for America'; put a comma after the word 'possible' and before the word 'and', as follows:  "We acknowledge the work of the Code for Sacramento and Open Fresno team, brigades of **C**ode for America, without whom this project would not have been possible, and acknowledge the input of the California Water Monitoring Council Trash Monitoring Workgroup."

Sentence 6--add comma after "In particular" and add the word "and" after "...Joseph Falkner, Democracy Lab,..." to make it clear that the names that follow are all associated with the University of Houston, as follows:  "In particular, we would like to acknowledge Joseph Falkner, Democracy Lab, **and** Brad Anderson, Jim Ewald, **and** Don Brower **of the** University of Houston. We acknowledge financial support from McPike Zima Charitable Foundation."

wincowgerDEV replied:
Sentence 4 recommendation made. https://github.com/code4sac/trash-ai/commit/21ad88f5d61a83cf08310bd2b93712986c5b13a9

Sentence 6, I might be a little confused are you saying Brad Anderson and Jim Ewald are with Democracy lab?
I am not sure if Don Brower is of the University of Houston, @walteryu can you confirm?

gdurante2019 replied:
Correction--after talking with Walter during the Code4Sac meetup this evening, please replace the original sentence with the following:

"In particular, we would like to acknowledge Joseph Falkner, Democracy Lab, Brad Anderson, Jim Ewald, Don Brower, and University of Houston.”

The only changes from the original sentence are the comma after "particular" and the word "and" before "University of Houston".

wincowgerDEV replied:
https://github.com/code4sac/trash-ai/commit/a582b32e159125913a9b0c0ab7bc996c04c9baa5

gdurante2019 reviewed (changes requested):
Please review and incorporate suggested changes, as outlined in the comments.

wincowgerDEV committed:
https://github.com/code4sac/trash-ai/pull/69#discussion_r1006489065

wincowgerDEV committed:
https://github.com/code4sac/trash-ai/pull/69#discussion_r1006514485

wincowgerDEV committed:
https://github.com/code4sac/trash-ai/pull/69#discussion_r1006517045

wincowgerDEV committed:
https://github.com/code4sac/trash-ai/pull/69#discussion_r1006519572

wincowgerDEV committed:
https://github.com/code4sac/trash-ai/pull/69#discussion_r1006536610

wincowgerDEV committed:
https://github.com/code4sac/trash-ai/pull/69#discussion_r1007668618

wincowgerDEV commented:
> Please review and incorporate suggested changes, as outlined in the comments.

All of your comments have been incorporated Gina :) Thanks for the great feedback.

wincowgerDEV committed:
add kris's comments

wincowgerDEV committed:
add WSL2 link

wincowgerDEV committed:
Update paper.md

wincowgerDEV committed:
add acknowledgements

wincowgerDEV commented:
@feydan I have gone through and made all requested changes to the manuscript. Could you merge the PR to main? Once it is merged to the main, I can go ahead and submit the manuscript :)

wincowgerDEV commented:
@gdurante2019 just making sure before we merge, did I address all your comments?

wincowgerDEV commented:
@shollingsworth not sure who has permission to merge this pull request but I believe everything has been addressed and it is ready to be merged.

wincowgerDEV committed:
Merge branch 'production' into manuscript

shollingsworth commented:
> @shollingsworth not sure who has permission to merge this pull request but I believe everything has been addressed and it is ready to be merged.

@gdurante2019 needs to re-review, once she hits approve we should be able to merge.

gdurante2019 commented on paper.md:
```diff
@@ -0,0 +1,113 @@
+Although computer vision classification routines have been created for trash, they have not been accessible to most researchers due to the challenges in deploying the models. Trash AI is a web GUI (Graphical User Interface) for serverless computer vision classification of individual items of trash within images, hosted at www.trashai.org. With a single batch upload and download, a user can automa ...
+
+# Statement of need
+The trash in the environment is a widespread problem that is difficult to measure. Policy makers require high quality data on trash to create effective policies. Classical measurement techniques require surveyors with pen and paper to manually quantify every piece of trash at a site. This method is time-consuming. Scientists are actively trying to address this issue by using imaging to better und ...
```
Delete "The" at the beginning of first sentence, and start the sentence with "Trash"

wincowgerDEV replied:
https://github.com/code4sac/trash-ai/commit/640e465f6b904b30308889c2a8fa56228e9c5fd0

gdurante2019 commented on paper.md:
```diff
@@ -0,0 +1,113 @@
+The AI model was developed starting with the TACO dataset, which was available with a complimentary Jupyter Notebook on [Kaggle](https://www.kaggle.com/datasets/kneroma/tacotrashdataset). An example notebook was referenced, which used the default `YOLO v5 model` [@Jocher:2020] as the basic model to begin transfer learning. Next, transfer learning was completed using the entire TACO dataset to imp ...
+
+## Limitations
+From our experience, the accuracy of the model varies depending on the quality of the images and their context/background. Trash is a word people use for an object that lacks purpose, and the purpose of an object is often not obvious in an image. Trash is a nuanced classification because the same object in different settings will not be considered trash (e.g., a drink bottle on someone's desk vs  ...
```
Second sentence, first word:  Put quotation marks around Trash, e.g., 'Trash' is a word people use...

wincowgerDEV replied:
https://github.com/code4sac/trash-ai/commit/3273b243ce38d7909f7aa8768abf9b017109c3ad

gdurante2019 reviewed (approved):
I've suggested a couple of minor edits as comments.  Please check those out and decide whether or not you'd like to incorporate.  Thanks!

wincowgerDEV committed:
https://github.com/code4sac/trash-ai/pull/69#discussion_r1033145379

wincowgerDEV committed:
https://github.com/code4sac/trash-ai/pull/69#discussion_r1033146187

wincowgerDEV committed:
https://github.com/code4sac/trash-ai/pull/69#discussion_r1024726101
